zipline/pipeline/loaders/blaze/core.py

```diff
@@ -1104,6 +1104,11 @@ def collect_expr(e, lower):
             materialized_deltas,
             dates,
         )
+        # If we ever have cases where we find out about multiple asof_dates'
+        # data on the same TS, we want to make sure that last_in_date_group
+        # selects the correct last asof_date's value.
+        sparse_output.sort_values(AD_FIELD_NAME, inplace=True)
```

Comment:
Then again, maybe having that extra, optional argument on that method would make people think about the sorting aspect when calling it. Either that, or we just add clarify in the docstring that the input df must be sorted so that the correct 'last' items are chosen. I think at this point I'm 50-50 on either option.